pkg/controllers/cleanup/controller.go

```diff
@@ -182,6 +182,23 @@ func (c *controller) cleanup(ctx context.Context, logger logr.Logger, policy kyv
 	debug := logger.V(4)
 	var errs []error
 
+	// get the deletion policy
+	deletionPolicy := spec.DeletionPropagationPolicy
+
+	// Add delete options based on the deletion policy
+	deleteOptions := &metav1.DeleteOptions{
+		PropagationPolicy: deletionPolicy,
+	}
+	// Check if the deletion policy is provided
+	if deletionPolicy != nil {
+		// If the user has provided a deletion policy, use it
+		deleteOptions.PropagationPolicy = deletionPolicy
+		logger.Info("Using specified deletion propagation policy", "policy", *deletionPolicy)
+	} else {
+		// Do not set a default value; let the API server decide
+		logger.Info("DeletionPropagationPolicy not provided, letting the API server decide")
+	}
```

Comment:
Why do we need this ?

Reply:
Updated the deletion policy logic to check if spec.DeletionPropagationPolicy is not nil and assign it directly to deleteOptions.PropagationPolicy.

Reply:
https://github.com/kyverno/kyverno/pull/11368/commits/c65a38d545f5d28d472c8e5f1cc05f5fb70cbf2c